### PR TITLE
Show Scala completions first in ScalaCli using directives

### PR DESCRIPTION
**Pull request by jkciesluk**
Now in `//> using lib "io.circe:@@"` scala completions (with adding second `:`) will show before java completions.
Also, fixes some issues e.g. showing completions after closing `"`

### CODE REVIEW

**Review by kpodsiad — COMMENTED** [review]
I'm confused because I see `fixes some issues` but there are no tests :)

File: mtags/src/main/scala-2/scala/meta/internal/pc/completions/ScalaCliCompletions.scala [hunk 1]
```diff
@@ -11,10 +11,12 @@ trait ScalaCliCompletions {
...
       path match {
         case Nil =>
-          CoursierComplete.isScalaCliDep(pos.lineContent.replace(CURSOR, ""))
+          CoursierComplete.isScalaCliDep(
+            pos.lineContent.replace(CURSOR, "").take(pos.column - 1)
+          )
         case (_: PackageDef) :: Nil if pos.source.file.path.endsWith(".sc") =>
           CoursierComplete.isScalaCliDep(
-            pos.lineContent.replace(CURSOR, "").take(pos.column)
+            pos.lineContent.replace(CURSOR, "").take(pos.column - 1)
```
Comment (tgodzik) [inline mtags/src/main/scala-2/scala/meta/internal/pc/completions/ScalaCliCompletions.scala:19]: Could you add the test case for it? of by one errors are a bit tricky.

File: mtags/src/main/scala/scala/meta/internal/mtags/CoursierComplete.scala [hunk 2]
```diff
@@ -62,8 +62,9 @@ object CoursierComplete {
     line match {
       case reg(deps) =>
         val dep =
-          deps.split(",").last.trim().stripPrefix("\"").stripSuffix("\"")
-        Some(dep)
+          deps.split(",").last
+        if (dep.endsWith("\"") || dep.endsWith(" ")) None
```
Comment (tgodzik) [inline mtags/src/main/scala/scala/meta/internal/mtags/CoursierComplete.scala:66]: Also a test case for this?

**Review by tgodzik — CHANGES_REQUESTED** [review]
Thanks for the fixes! Let's maybe add some test cases to show what was fixed.

File: mtags/src/main/scala-2/scala/meta/internal/pc/completions/ScalaCliCompletions.scala (hunk 1 above)
Comment (jkciesluk) [reply mtags/src/main/scala-2/scala/meta/internal/pc/completions/ScalaCliCompletions.scala:19]: Actually it's not an error, but I haven't realised before that `pos.column` in Scala 2 starts counting from 1 😮 not from 0 like in Scala 3. So in `ab@@cd` `pos.column == 3`, hence the `- 1`. Should I add a comment in the code?

File: mtags/src/main/scala/scala/meta/internal/mtags/CoursierComplete.scala (hunk 2 above)
Comment (jkciesluk) [reply mtags/src/main/scala/scala/meta/internal/mtags/CoursierComplete.scala:66]: I've added the tests for this

**Review by tgodzik — APPROVED** [review]

**Pull request merged**